### PR TITLE
Update dev docs for consistency

### PR DESCRIPTION
**Pull request by rami-a**
This change updates the following component dev docs to align better:

- Date pickers
- Progress indicators
- Radio buttons
- Sliders
- Tabs
- Text fields
- Time pickers

### CODE REVIEW

File: docs/components/date_pickers.md [hunk 1]
```diff
@@ -34,6 +33,10 @@ Date pickers can be embedded into:
 
 Flutter's APIs support accessibility setting for large fonts, screen readers, and sufficient contrast. For more information, go to Flutter's [accessibility](https://flutter.dev/docs/development/accessibility-and-localization/accessibility) and [internationalization](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) pages.
 
+### Types
+
+There are two types of date pickers: 1\. [date pickers](#date-pickers) 2\. [date range pickers](#date-range-pickers)
```
Comment (clocksmith) [inline docs/components/date_pickers.md:38]: what is the \ for?

File: docs/components/text_fields.md [hunk 2]
```diff
@@ -32,10 +32,7 @@ For more guidance on writing labels, go to [our page on how to write a good acce
 
 ### Types
 
-The text field class consists of the following types:
-
-- [Filled text](#filled-text)
-- [Outlined text](#outlined-text)
+The text field class consists of the following types: 1\. [filled text](#filled-text) 2\. [outlined text](#outlined-text)
```
Comment (guidezpl) [inline docs/components/text_fields.md]: nit:
```suggestion
There are two types of text fields: 1\. [filled text](#filled-text) 2\. [outlined text](#outlined-text)
```

**Review by clocksmith — APPROVED** [review]

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

File: docs/components/date_pickers.md (hunk 1 above)
Comment (rami-a) [reply docs/components/date_pickers.md:38]: I believe thats what makes it render as a list on material.io

See https://material.io/components/tabs/flutter#using-tabs for example

File: docs/components/text_fields.md (hunk 2 above)
Comment (rami-a) [reply docs/components/text_fields.md]: Good idea, done

**Pull request merged**